### PR TITLE
Improve removal of nested MODX tag content in sanitizeRequest

### LINKED ISSUE #15366: sanitizeRequest doesn't remove MODX tags completely
### Summary
It seems santizeRequest doesn't remove all MODX tags completely

### Step to reproduce
Try to save a form from fronend with something like that:

```
[[test? 
  &ids = `[[!getids 
    &field=`id` 
    &resource=`[[+resource]]`
    ]]
  &parents=`2`
`]]
```

### Observed behavior
this keeps that:

```
`
    
  &parents=`2`
`
```

### Expected behavior
It should remove everything within the MODX tag completely

### Related issue(s)/PR(s)
Let us know if this is related to any issue/pull request.

### Environment
MODX 2.8.1

changing in modx.class.php

```
    public $sanitizePatterns = array(
        'scripts'       => '@<script[^>]*?>.*?</script>@si',
        'entities'      => '@&#(\d+);@',
        'tags1'          => '@\[\[(.*?)\]\]@si',
        'tags2'          => '@(\[\[|\]\])@si',
    );
```

to

```
    public $sanitizePatterns = array(
        'scripts'       => '@<script[^>]*?>.*?</script>@si',
        'entities'      => '@&#(\d+);@',
        'tags1'          => '@\[\[(?:(?!(\[\[|\]\])).)*\]\]@si',
        'tags2'          => '@(\[\[|\]\])@si',
    );
```
seems to solve that

### PR DESCRIPTION
**Pull request by Bruno17**
### What does it do?
Fix for #15366

### Why is it needed?
See #15366

### How to test
See #15366

### Related issue(s)/PR(s)
#15366

### CODE REVIEW

**Comment by JoshuaLuckers** [comment]
It would be nice to have this covered in the Unit Test: https://github.com/modxcms/revolution/blob/5b912f57d92de3bb0ac94923af06686f05d4896d/_build/test/Tests/Model/Request/modRequestTest.php#L298

A new test has to be added in the in providerSanitizeRequest method in that test class.

**Comment by Bruno17** [comment]
ok, have added a test 
...... and have signed a CLA long time ago and contributed in the past

**Comment by Ibochkarev** [comment]
@cla-bot

**Comment by opengeek** [comment]
> ok, have added a test
> ...... and have signed a CLA long time ago and contributed in the past

@Bruno17 — can you check and make sure your GitHub username is correct at the CLA link there?

**Comment by Bruno17** [comment]
@cla-bot check

File: _build/test/Tests/Model/Request/modRequestTest.php [hunk 1]
```diff
@@ -316,7 +316,14 @@ public function providerSanitizeRequest() {
             array('MODX [[fakeSnippet]] Tags','MODX  Tags'),
             array("MODX [[\$chunk? &property=`test`\n &across=`lines
 
-` &test=1]] Tags",'MODX  Tags'),
+                   `&test=1]] Tags",'MODX  Tags'),
+            array("Nested MODX [[test? 
+                   &ids = `[[!getids 
+                   &field=`id` 
+                   &resource=`[[+resource]]`
+                   ]]
```
Comment (Jako) [inline _build/test/Tests/Model/Request/modRequestTest.php:324]: The test even works with invalid tags. A ` is missing here.

Comment (JoshuaLuckers) [reply _build/test/Tests/Model/Request/modRequestTest.php:324]: @Jako can you submit a PR with a fix for the test?

Comment (Jako) [reply _build/test/Tests/Model/Request/modRequestTest.php:324]: It makes no difference. Since the ` does not touch the regular expression for nested tags.

**Pull request merged**